File: greatalovesvini-main/PeriodTracker.Model/Entities/CycleSymptom.cs
using System.Text.Json.Serialization;
using PeriodTracker.Model.Entities;
using System;

public class CycleSymptom
{
    public CycleSymptom(int id)
    {
        CycleSymptomId = id;
    }

    [JsonConstructor]
    public CycleSymptom()
    {
    }

    public int CycleSymptomId { get; set; }
    public int CycleId { get; set; }
    public int SymptomId { get; set; }
    public int Intensity { get; set; }
    public DateTime Date { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.Now;
    public PeriodCycle PeriodCycle { get; set; }
    public Symptom Symptom { get; set; }
}

File: Frontend/src/app/models/CycleSymptom.ts
import { Symptom } from "./Symptom";

/**
 * CycleSymptom Model - Maps symptoms to period cycles
 * 
 * This interface represents a symptom experienced during a specific cycle,
 * creating a many-to-many relationship between cycles and symptoms.
 */
export interface CycleSymptom {
  cycleSymptomId: number;  // Primary key
  cycleId: number;         // Foreign key to cycle
  symptomId: number;       // Foreign key to symptom
  intensity: number;       // Severity rating (1-5)
  date: string;            // Date symptom was experienced
  createdAt: string;       // Record creation timestamp
  symptom?: Symptom;       // Optional nested symptom object
}

/**
* Data transfer object for creating new cycle symptoms
* Simplified version of CycleSymptom for API requests
*/
export interface CreateCycleSymptomDto {
  cycleId: number;         // Which cycle
  symptomId: number;       // Which symptom
  intensity: number;       // How severe (1-5)
  date: string;            // When experienced (YYYY-MM-DD)
}